sap-xmake-build.sh

```diff
@@ -0,0 +1,219 @@
+
+DATE_STRING=`date +%Y%m%d%H%M%S`
+GIT_REPO="https://github.com/Altiscale/oozie"
+
+export RPM_NAME=`echo alti-oozie-client-${OOZIE_VERSION}`
+export RPM_DESCRIPTION="Apache Oozie Client ${OOZIE_VERSION}\n\n${DESCRIPTION}"
+export RPM_DIR="${RPM_DIR:-"${INSTALL_DIR}/oozie-client-artifact/"}"
+mkdir --mode=0755 -p ${RPM_DIR}
+
+echo ""
+echo "Current Directory: $(pwd)"
+
+echo ""
+echo "[INFO] Packaging oozie client rpm ${RPM_DIR}${RPM_NAME}-${ALTISCALE_RELEASE}-${DATE_STRING}.x86_64.rpm" 
+
+# Make the Client RPM
+cd ${RPM_DIR}
+
+fpm --verbose \
+--maintainer ops@altiscale.com \
+--vendor Altiscale \
+--provides ${RPM_NAME} \
+-s dir \
+-t rpm \
+-n ${RPM_NAME} \
+-v ${ALTISCALE_RELEASE} \
+--epoch 1 \
+--description "${DESCRIPTION}" \
+--iteration ${DATE_STRING} \
+--rpm-user root \
+--rpm-group root \
+-C ${INSTALL_DIR} \
+opt
+
+echo "[INFO] Finished packaging oozie client rpm ${RPM_DIR}${RPM_NAME}-${ALTISCALE_RELEASE}-${DATE_STRING}.x86_64.rpm" 
+echo ""
+
+echo "[INFO] find . -name *.rpm -print in $(pwd)"
+find . -name *.rpm -print
+echo ""
+
+echo "[INFO] cp ${RPM_DIR}${RPM_NAME}-${ALTISCALE_RELEASE}-${DATE_STRING}.x86_64.rpm ${RPM_DIR}alti-oozie-client-${XMAKE_PROJECT_VERSION}.rpm"
+cp "${RPM_DIR}${RPM_NAME}-${ALTISCALE_RELEASE}-${DATE_STRING}.x86_64.rpm" "${RPM_DIR}alti-oozie-client-${XMAKE_PROJECT_VERSION}.rpm"
+echo ""
+
+# Make the Server RPM
+echo "[INFO] Making oozie server rpm"
+INSTALL_DIR="$MY_DIR/oozieserverrpmbuild"
+echo "[INFO] mkdir -p --mode=0755 ${INSTALL_DIR}"
+mkdir --mode=0755 -p ${INSTALL_DIR}
+
+export DEST_DIR=${INSTALL_DIR}/opt
+rm -rf ${DEST_DIR}
+echo "[INFO] mkdir -p --mode=0755 ${DEST_DIR}"
+mkdir -p --mode=0755 ${DEST_DIR}
+echo ""
+
+cd ${DEST_DIR}
+echo "[INFO] tar -xvzpf ${MY_DIR}/distro/target/oozie-${OOZIE_VERSION}-distro.tar.gz into ${DEST_DIR}"
+tar -xvzpf ${MY_DIR}/distro/target/oozie-${OOZIE_VERSION}-distro.tar.gz
+echo ""
+
+export RPM_NAME=`echo alti-oozie-server-${OOZIE_VERSION}`
+export RPM_DESCRIPTION="Apache Oozie Server ${OOZIE_VERSION}\n\n${DESCRIPTION}"
+export RPM_DIR="${INSTALL_DIR}/oozie-server-artifact/"
+echo "[INFO] mkdir -p --mode=0755 ${RPM_DIR}"
+mkdir -p --mode=0755 ${RPM_DIR}
+
+export RPM_BUILD_DIR="${INSTALL_DIR}/opt/oozie-${OOZIE_VERSION}"
+echo "[INFO] mkdir -p --mode=0755 ${RPM_BUILD_DIR}"
+mkdir -p  --mode=0755 ${RPM_BUILD_DIR}
+echo "[INFO] mkdir -p --mode=0775 ${RPM_BUILD_DIR}/libext"
+mkdir -p --mode=0775 ${RPM_BUILD_DIR}/libext
+echo ""
+
+echo "[INFO] cp ${MY_DIR}/distro/target/oozie-${OOZIE_VERSION}-distro/oozie-${OOZIE_VERSION}/oozie.war ${RPM_BUILD_DIR}"
+cp ${MY_DIR}/distro/target/oozie-${OOZIE_VERSION}-distro/oozie-${OOZIE_VERSION}/oozie.war ${RPM_BUILD_DIR} 
+
+cd ${RPM_BUILD_DIR}/libext
+echo "[INFO] cp /imports/sap-ext-js-2.2.zip ${RPM_BUILD_DIR}/libext/ext-2.2.zip"
+cp /imports/sap-ext-js-2.2.zip ./ext-2.2.zip
+
+cd ${RPM_BUILD_DIR}/conf
+rm -rf hadoop-conf
+ln -s /etc/hadoop hadoop-conf
```

Comment:
Just a note here as a reminder. I did a precautious study on the existing Oozie rpm and it deploys the file `/opt/oozie-client-4.2.0/conf/hadoop-conf/core-site.xml`. So if this directory is an external dependency where as the directory `/opt/oozie-client-4.2.0/conf/hadoop-conf` is created by another app cookbook (e.g. hadoop app cookbook? or chef-solo), I will keep an eye whether this file will still be there when the Oozie rpm is created by xMake from this PR.

```
[aleefoa@desktop-foa ~]$ rpm -ql alti-oozie-client-4.2.0-4.2.0-201606162101.x86_64 | grep conf
/opt/oozie-client-4.2.0/conf/action-conf/hive.xml
...
/opt/oozie-client-4.2.0/conf/hadoop-conf/core-site.xml
/opt/oozie-client-4.2.0/conf/hadoop-config.xml
/opt/oozie-client-4.2.0/conf/oozie-default.xml.reference
/opt/oozie-client-4.2.0/conf/oozie-env.cmd
...
```

If the RPM generated from xMake does not include that file `/opt/oozie-client-4.2.0/conf/hadoop-conf/core-site.xml`, then we perhaps want to verify whether app cookbook will deploy it to fill the gap. @anil-altiscale  you may want to check hadoop app cookbook to see if it is doing some magic there, or is this a pure Oozie logic that copies over a file from `/etc/hadoop`. Since Oozie is a legacy product, so I will perhaps pay a little more attention on these internal dependencies and orders :(

Reply:
@alee-altiscale I have verified that this file is there in the new RPM. I have also done a diff between old and new rpms to spot exactly this - any cases of missing files or packages...

Reply:
Awesome ~ thanks for confirming.